### PR TITLE
Fix Callout sorting

### PR DESCRIPTION
**Pull request by ccanos**
Goes together with https://github.com/alkem-io/server/pull/3120
### Describe the background of your pull request

What does your pull request do? Does it solve a bug (which one?), add a feature?

### Additional context

Add any other context

### Governance 

- [ ] Documentation is added
- [ ] Test cases are added or updated

By submitting this pull request I confirm that:
- I've read the contributing document https://github.com/alkem-io/.github/blob/master/CONTRIBUTING.md.
- I've read and understand the Code of Conduct https://github.com/alkem-io/.github/blob/master/CODE_OF_CONDUCT.md.
- I understand that any contributions or suggestions I made may make it into the actual code. I've read the License 
     https://github.com/alkem-io/Coordination/blob/master/LICENSE.
- I understand that submitting the pull request requires agreeing to and signing the contributor license agreement.

### CODE REVIEW

File: src/core/utils/UpdateOrder.ts [hunk 1]
```diff
@@ -1,31 +1,19 @@
 import { without } from 'lodash';
-import { Identifiable } from '../../domain/shared/types/Identifiable';
 
 export interface OrderUpdate {
   (ids: string[]): string[];
 }
 
 interface Insert {
-  (nextItems: string[], item: string, index: number): void;
```
Comment (ccanos) [inline src/core/utils/UpdateOrder.ts:9]: removed index calculation because it was not used anywhere

File: src/core/utils/UpdateOrder.ts [hunk 2]
```diff
@@ -1,31 +1,19 @@
...
-    const index = prevItems.indexOf(item);
     const nextItems = without(prevItems, item);
-    insert(nextItems, item, index);
+    insert(nextItems, item);
     return nextItems;
   });
 };
 
 export default UpdateOrder;
-
```
Comment (ccanos) [inline src/core/utils/UpdateOrder.ts:22]: moved to CalloutsView.tsx

File: src/domain/collaboration/InnovationFlow/utils/sortCallouts.spec.ts [hunk 3]
```diff
@@ -38,8 +38,8 @@ describe('sortCallouts handling grouped Callouts', () => {
         newState: FlowStates.C,
       }).groupedCalloutIds
     ).toEqual({
-      [FlowStates.A]: ['1', '5', '2'],
-      [FlowStates.B]: ['6', '3'],
+      // [FlowStates.A]: ['1', '5', '2'],
```
[the remark this answers is not in the document]
Comment (ccanos) [reply src/domain/collaboration/InnovationFlow/utils/sortCallouts.spec.ts:41]: I would like to leave them... I don't think we're going to change back the behavior, but this way we know what's on the other groups if we modify the tests in the future... 
I don't know, these are just tests, I thought it was good to keep them but I don't really mind

File: src/core/utils/UpdateOrder.ts [hunk 4]
```diff
@@ -20,12 +19,38 @@ const UpdateOrder = (setItems: (update: OrderUpdate) => void) => (insert: Insert
...
-  if (targetCalloutIndex === -1) {
+/**
+ * We need to find the callout that we are moving in the list of all callouts loaded.
+ * Then see its position between the callouts in the same group.
+ * Find the previous or next callout in that group. And then find the position of that one in the list of all callouts.
+ */
+export const findTargetItemIndex = (
+  position: 'prev' | 'next',
+  allCallouts: CalloutWithDisplayLocation[] | undefined,
+  nextIds: string[],
```
[the remark this answers is not in the document]
Comment (ccanos) [reply src/core/utils/UpdateOrder.ts]: refactored fully

**Comment by valentinyanakiev** [comment]
Tested locally and worked very well for me.

**Pull request merged**